src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
```

Comment:
The initial state is cloned correctly here, but ensure that each action results in a new clone of the state to avoid unintended mutations.